labelbox/schema/foundry.py

```diff
@@ -0,0 +1,160 @@
+MODEL_FIELD_NAMES = list(Model.schema()['properties'].keys())
+
+
+class FoundryClient:
+
+    def __init__(self, client):
+        self.client = client
+
+    def _create_app(self, app: App) -> App:
+        query_str = """
+            mutation CreateDataRowAttachmentPyApi(
+                $name: String!, $modelId: ID!, $description: String, $inferenceParams: Json!, $classToSchemaId: Json!
+            ){
+                createModelFoundryApp(input: {
+                    name: $name 
+                    modelId: $modelId 
+                    description: $description
+                    inferenceParams: $inferenceParams
+                    classToSchemaId: $classToSchemaId
+                })
+                {
+                    id
+                    name
+                    modelId
+                    description
+                    inferenceParams
+                    classToSchemaId
+                }
+            }
+            """
+
+        params = app.dict(by_alias=True, exclude={"id"})
+
+        response = self.client.execute(query_str, params)
+        return App(**response["createModelFoundryApp"])
+
+    def _get_app(self, id: str) -> App:
+        field_names_str = "\n".join(APP_FIELD_NAMES)
+
+        query_str = f"""
+            query GetFoundryAppByIdPyApi($id: ID!) {{
+                findModelFoundryApp(where: {{id: $id}}) {{
+                    {field_names_str}
+                }}
+            }}
+        """
+        params = {"id": id}
+
+        try:
+            response = self.client.execute(query_str, params)
+        except exceptions.InvalidQueryError as e:
+            raise exceptions.ResourceNotFoundError(App, params)
+        return App(**response["findModelFoundryApp"])
+
+    def _delete_app(self, id: str) -> None:
+        query_str = """
+            mutation DeleteFoundryAppPyApi($id: ID!) {
+                deleteModelFoundryApp(id: $id) {
+                    success
+                }
+            }
+        """
+        params = {"id": id}
+        self.client.execute(query_str, params)
+
+    def _get_model(self, id: str) -> Model:
+        field_names_str = "\n".join(MODEL_FIELD_NAMES)
+
+        query_str = f"""
+            query GetModelByIdPyApi($id: ID!) {{
+                modelFoundryModel(where: {{id: $id}}) {{
+                    model {{
+                        {field_names_str}
+                    }}
+                }}
+            }}
+        """
+        params = {"id": id}
+
+        response = self.client.execute(query_str, params)
+        return Model(**response["modelFoundryModel"]["model"])
+
+    def run_app(self, model_run_name: str,
+                data_rows: Union[DataRowIds, GlobalKeys], app_id: str) -> Task:
+        app = self._get_app(app_id)
+        model = self._get_model(app.model_id)
+
+        if isinstance(data_rows, DataRowIds):
+            data_rows_query = {
+                "type": "data_row_id",
```

Comment:
suggestion: if you could change the type name, you could then use `IdType` and make this code more flexible. IdType(s) names are the same across API and ADV btw. Perhaps it would be good to stick to the same naming conventions?

Reply:
This is the query format used in a large part of api codebase and not easy to change.